src/hotspot/share/runtime/os.hpp

```diff
@@ -197,7 +197,7 @@ class os: AllStatic {
 
   static char*  pd_reserve_memory(size_t bytes, bool executable);
 
-  static char*  pd_attempt_reserve_memory_at(char* addr, size_t bytes, bool executable);
+  static char*  pd_attempt_reserve_memory_at(char* addr, size_t bytes, bool executable, int *fail_reason);
```

Comment:
If `fail_reason` defaults to `nullptr` you would save making a few changes above.